queries.sql

```diff
@@ -71,17 +71,70 @@ SELECT species.name, COUNT(*) FROM animals
     GROUP BY species.id;
 
 SELECT animals.name, species.name FROM animals
-    INNER JOIN owners ON owners.id = animals.owner_id
-    INNER JOIN species ON species.id = animals.species_id
+    JOIN owners ON owners.id = animals.owner_id
+    JOIN species ON species.id = animals.species_id
     WHERE owners.full_name = 'Jennifer Orwell'
     AND species.name = 'Digimon';
 
 SELECT animals.name, animals.escape_attempts FROM animals
-    INNER JOIN owners ON owners.id = animals.owner_id
+    JOIN owners ON owners.id = animals.owner_id
     WHERE owners.full_name = 'Dean Winchester'
     AND animals.escape_attempts = 0;
 
 SELECT owners.full_name, COUNT(animals.owner_id) FROM animals 
     FULL OUTER JOIN owners 
     ON animals.owner_id = owners.id
-    GROUP BY owners.id;
\ No newline at end of file
+    GROUP BY owners.id;
+
+/* Add join table for visits */
+
+SELECT animal_id, visits.date_of_visit FROM visits
+    JOIN animals ON animal_id = visits.animal_id
+    WHERE vet_id = 1
+    ORDER BY date_of_visit DESC LIMIT 1;
+
+SELECT COUNT(DISTINCT visits.animal_id) FROM visits
+    JOIN vets ON vets.id = visits.vet_id
+    WHERE vets.name = 'Stephanie Mendez';
+
+SELECT vets.name, species.name FROM vets 
+    LEFT JOIN specializations ON specializations.vet_id = vets.id
+    LEFT JOIN species ON species.id = specializations.species_id;
+
+SELECT animals.name, visits.date_of_visit FROM visits
+    JOIN vets ON vets.id = visits.vet_id
+    JOIN animals ON animals.id = visits.animal_id
+    WHERE vets.name = 'Stephanie Mendez'
+    AND visits.date_of_visit BETWEEN 'Apr 1, 2020' AND 'Aug 30, 2020';
+
+SELECT animals.name, COUNT(visits.animal_id) FROM visits
+    JOIN vets ON vets.id = visits.vet_id
+    JOIN animals ON animals.id = visits.animal_id
+    GROUP BY animals.name, visits.animal_id
+    ORDER BY COUNT(visits.animal_id) DESC LIMIT 1;
+
+SELECT animals.name, visits.date_of_visit FROM visits
+    JOIN vets ON vets.id = visits.vet_id
+    JOIN animals ON animals.id = visits.animal_id
+    WHERE vets.name = 'Maisy Smith'
+    GROUP BY animals.name, visits.date_of_visit
+    ORDER BY visits.date_of_visit LIMIT 1;
+
+SELECT * FROM visits
+    JOIN animals ON animals.id = visits.animal_id
+    JOIN vets ON vets.id = visits.vet_id
+    ORDER BY visits.date_of_visit DESC LIMIT 1;
+
+SELECT COUNT(visits.animal_id) FROM visits
+    JOIN vets ON vets.id = visits.vet_id
+    JOIN animals ON animals.id = visits.animal_id
+    JOIN specializations ON specializations.vet_id = vets.id
+    WHERE specializations.species_id <> animals.species_id;
+
+SELECT species.name, COUNT(visits.animal_id) FROM visits
+    JOIN vets ON vets.id = visits.vet_id
+    JOIN animals ON animals.id = visits.animal_id
+    JOIN species ON species.id = animals.species_id
+    WHERE vets.name = 'Maisy Smith'
+    GROUP BY species.name
+    ORDER BY COUNT(visits.animal_id) DESC LIMIT 1;
```

Comment:
- [ ] Please add a query for this question `Who was the last animal seen by William Tatcher?` I can't find it

Reply:
Thank you for the review. The query you're asking for is the first one however it returned the animal id, I have adjusted it to return the animal's name instead